content/blog/2020/08/2020-08-29-git-performance-improvement-phase3.adoc

```diff
@@ -0,0 +1,126 @@
+
+Building Tensorflow (~800 MiB) using a Jenkins pipeline, there is over 50% reduction in overall time spent in completing a job! 
+The result is consistent multiple platforms.
+
+The reason for such a decrease is the fact that `JGit` degrades in performance when we are talking about large sized repositories. Since the GitToolChooser is aware of this fact, it chooses to recommend `command line git` instead which saves the user some time.
+
+== Minor Performance Improvements
+
+*Note: Enable JGit before using the new performance features to let GitToolChooser work with more options* -> link:https://plugins.jenkins.io/git-client/#enabling-jgit[Here's how]
+
+image:/images/post-images/2020/07-git-plugin-performance-improvement/git-minor-perf.png[title="Minor performance enhancements" role="center", width=400]
+
+Building the git plugin (~ 20 MiB) using a Jenkins pipeline, there is a drop of a *second* across all platforms when performance enhancement is enabled. Also, eliminating a redundant fetch reduces unnecessary load on git servers.
+
+The reason for this change is the fact that `JGit` performs better than `command line git` for small sized repositories (<50MiB) as an already warmed up JVM favors the native Java implementation.
+
+== Release
+
+* Git Plugin 4.4.0 -> https://github.com/jenkinsci/git-plugin/releases/tag/git-4.4.0
+  ** Add GitToolChooser
+  ** Remove redundant fetch
+* Git Client Plugin 3.4.0 -> https://github.com/jenkinsci/git-client-plugin/releases/tag/git-client-3.4.0
+  ** Add support to communicate compatibility of JGit with certain additional SCM behaviors
+
+== The Road Ahead
+
+* Support from other branch source plugins
+  ** Plugins like the GitHub Branch Source Plugin or GitLab Branch Source Plugin need to extend an extension point provided by the git plugin to facilitate the exchange of information related to size of a remote repository hosted by the particular git provider
+* JENKINS-63519: GitToolChooser predicts the wrong implementation
+* Addition of this feature to GitSCMSource
+* Detection of lock related delays accessing the cache directories present on the controller
+  ** This issue was reported by the plugin maintainer Mark Waite, there is a need to reproduce the issue first and then find a possible solution.
+
+== Reaching out
+
+Feel free to reach out to us for any questions or feedback on the project's
+link:https://gitter.im/jenkinsci/git-plugin[Gitter Channel] or the
+mailto:jenkinsci-dev@googlegroups.com[Jenkins Developer Mailing list].
+Report an issue at Jenkins link:https://issues.jenkins-ci.org/[Jira].
+
+== Useful Links
+
+* Phase 1 Blog: https://www.jenkins.io/blog/2020/07/09/git-performance-improvement-phase1/
+* Phase 2 Blog: https://www.jenkins.io/blog/2020/07/29/git-performance-improvement-phase2/
+* Project Page: https://www.jenkins.io/projects/gsoc/2020/projects/git-plugin-performance/
+* Final Demo Presentation: https://www.youtube.com/watch?v=sEqK4dAFg4s&feature=youtu.be
```

Comment:
Embed the video into the blog post so that it is immediately visible as a video.  Link to the other posts rather than making the links visible

```suggestion
== Useful links

* link:/blog/2020/07/09/git-performance-improvement-phase1/[Phase 1 Blog]
* link:/blog/2020/07/29/git-performance-improvement-phase2/[Phase 2 Blog]
* link:/projects/gsoc/2020/projects/git-plugin-performance/[Project Page]

=== Demonstration

video::sEqK4dAFg4s[youtube,width=800,height=420]
```